src/FastSerialization.Tests/SegmentedDictionaryTests.cs

```diff
@@ -0,0 +1,229 @@
+            
+            Assert.Equal(3, dict.Count);
+            
+            bool removed = dict.Remove(2);
+            Assert.True(removed);
+            Assert.Equal(2, dict.Count);
+            Assert.False(dict.ContainsKey(2));
+            
+            removed = dict.Remove(99);
+            Assert.False(removed);
+            Assert.Equal(2, dict.Count);
+        }
+
+        [Fact]
+        public void Clear()
+        {
+            var dict = new SegmentedDictionary<string, int>();
+            dict.Add("a", 1);
+            dict.Add("b", 2);
+            dict.Add("c", 3);
+            
+            Assert.Equal(3, dict.Count);
+            
+            dict.Clear();
+            Assert.Empty(dict);
+            Assert.False(dict.ContainsKey("a"));
+        }
+
+        [Fact]
+        public void UpdateValue()
+        {
+            var dict = new SegmentedDictionary<string, int>();
+            dict.Add("key", 10);
+            Assert.Equal(10, dict["key"]);
+            
+            dict["key"] = 20;
+            Assert.Equal(20, dict["key"]);
+        }
+
+        [Fact]
+        public void AddDuplicateKeyThrows()
+        {
+            var dict = new SegmentedDictionary<string, int>();
+            dict.Add("duplicate", 1);
+            
+            Assert.Throws<ArgumentException>(() => dict.Add("duplicate", 2));
+        }
+
+        [Fact]
+        public void AccessNonExistentKeyThrows()
+        {
+            var dict = new SegmentedDictionary<string, int>();
+            
+            Assert.Throws<KeyNotFoundException>(() => { var x = dict["missing"]; });
+        }
+
+        [Fact]
+        public void Keys()
+        {
+            var dict = new SegmentedDictionary<string, int>();
+            dict.Add("one", 1);
+            dict.Add("two", 2);
+            dict.Add("three", 3);
+            
+            var keys = dict.Keys.ToList();
+            Assert.Equal(3, keys.Count);
+            Assert.Contains("one", keys);
+            Assert.Contains("two", keys);
+            Assert.Contains("three", keys);
+        }
+
+        [Fact]
+        public void Values()
+        {
+            var dict = new SegmentedDictionary<string, int>();
+            dict.Add("one", 1);
+            dict.Add("two", 2);
+            dict.Add("three", 3);
+            
+            var values = dict.Values.ToList();
+            Assert.Equal(3, values.Count);
+            Assert.Contains(1, values);
+            Assert.Contains(2, values);
+            Assert.Contains(3, values);
+        }
+
+        [Fact]
+        public void GetEnumerator()
+        {
+            var dict = new SegmentedDictionary<string, int>();
+            dict.Add("one", 1);
+            dict.Add("two", 2);
+            dict.Add("three", 3);
+            
+            int count = 0;
+            foreach (var kvp in dict)
+            {
+                Assert.NotNull(kvp.Key);
+                Assert.True(kvp.Value >= 1 && kvp.Value <= 3);
```

Comment:
Please modify this to be more explicit.  It should find each of the entries exactly once and make sure the key and value are correct.